src/parser.rs

```diff
@@ -626,35 +625,80 @@ impl<'a> Parser<'a> {
 
           attributes.push(Attribute { start, key, value });
         }
-        Some((_, '/')) if matches!(markup_kind, MarkupKind::Open) => {
+        Some((self_close, '/')) => {
+          if matches!(markup_kind, MarkupKind::Close) {
+            self.report(Diagnostic::MarkupCloseInvalidSelfClose {
+              self_close_loc: self_close,
+            });
+          }
           self.next(); // consume '/'
           markup_kind = MarkupKind::Standalone;
-          let Some(_) = self.eat('}') else { panic!() };
-          break;
+          match self.peek() {
+            Some((_, '}')) => {
+              self.next(); // consume '}'
+              break false;
+            }
+            Some((before_first_space, c)) if is_space(c) => {
+              self.skip_spaces();
+              match self.peek() {
+                Some((close_brace, '}')) => {
+                  self.next(); // consume '}'
+                  self.report(
+                    Diagnostic::MarkupInvalidSpaceBetweenSelfCloseAndBrace {
+                      space: Span::new(before_first_space..close_brace),
+                    },
+                  );
+                  break false;
+                }
+                None => {
+                  break true;
+                }
+                _ => {
+                  self.text.reset_to(before_first_space);
+                  todo!("report expected }} after /")
```

Comment:
```suggestion
                  todo!("report expected } after /")
```

Reply:
The first `}` escapes the second, because this is a `format!` style macro: https://doc.rust-lang.org/std/fmt/index.html#escaping